src/gui/callstatechecker.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2022 by Claudio Cambra <claudio.cambra@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonArray>
+
+#include "callstatechecker.h"
+#include "account.h"
+
+namespace OCC {
+
+Q_LOGGING_CATEGORY(lcCallStateChecker, "nextcloud.gui.callstatechecker", QtInfoMsg)
+
+constexpr int successStatusCode = 200;
+
+CallStateChecker::CallStateChecker(QObject *parent)
+    : QObject(parent)
+{
+    setup();
+}
+
+void CallStateChecker::setup()
+{
+    _notificationTimer.setSingleShot(true);
+    _notificationTimer.setInterval(60 * 1000);
+    connect(&_notificationTimer, &QTimer::timeout, this, &CallStateChecker::slotNotificationTimerElapsed);
+
+    _statusCheckTimer.setInterval(2 * 1000);
```

Comment:
Since this basically DDoSes the server, you might use 5 secs as well like the mobile clients do?
https://github.com/nextcloud/talk-ios/blob/529bdc88341518569a7a470b6ca7a535d6e70cdf/NextcloudTalk/CallKitManager.m#L44

But also in case the user joined, you should receive a notify_push about the call notification being deleted.
Listening to that and then removing also the popup+sound would have been good enough from my POV, but now you made the effort already so it's okay to keep.